docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,434 @@
+
+Devices in manual management mode are not managed by the fleet controller, the device
+controller will not attempt to reconcile the device spec with the fleet spec.
+
+In this mode, the device.spec is fully writtable through this API.
+
+If the device is selected by a fleet, the `spec.managementMode` can be flipped to `fleet`, and the
+device controller will start managing the device again.
+
+While devices are managed by fleets, all floating tags are resolved to stable hashes, but
+when a device is managed manually, the floating tags can be used, see the next sub-section
+for additional details.
+
+###### Expected behaviour regarding external references with floating tags
+
+Changes on external references is monitored and acted on (git branches,
+device images, secrets), the device controller should re-evaluate the device
+spec and apply the changes to the rendered version which will be exposed
+on the `/device/{name}/rendered` endpoint.
+
+Several mechanisms can be used to track external references:
+* Polling
+* Webhooks
+
+#### `/device/{name}/rendered` (device side)
+This endpoint is used by the device agent to retrieve the rendered device spec, this endpoint is
+read-only, each device can only access it's own device rendered spec based on the device
+certificate identification, an administrator can access any device rendered spec for debug purposes.
+
+The agent can pass two parameters: `version`and `owner` if those haven't changed the API will return
+204 No Content, if the version or fleet has changed the API will return the rendered spec.
+
+Change proposed:
+* The parameter name `fleet` is changed to `owner` to be consistent with the metadata field.
+
+Behavior changes:
+* If a device configuration is not yet rendered for some reason, FC will queue the device for rendering and instruct
+  the device to try again later by returning 202 Accepted.
+
+The administrator can find the fully rendered device spec by using the following cmdline command:
+```shell
+flightctl get device/xxxxxxx --rendered
+```
+
+The JSON blob configuration exposed to the device in the /device/{name}/rendered endpoint
+while compacted for transport, should be formated in a way that it's easy to read by humans
+when printed out to the console, by adding tabs and newlines.
+
+#### `/device/{name}/status` (device side)
+
+This endpoint is used by the device agent to report the status of the device, this endpoint is write-only,
+each device can only access it's own device status based on the device certificate identification.
+
+The result of this write will be exposed on the `/device/{name}` endpoint as part of the status field.
+
+Reporting to this endpoint also serves as a heartbeat mechanism, the device agent should report the status,
+if a device stops reporting to this endpoint in the expected timelines will be considered offline.
+
+NOTE: When optimization is required in the future we could implement an additional endpoint to report heartbeats,
```

Comment:
There may be other ways, e.g. implementing a delta update. Not sure we should detail one possible option here?

Reply:
👍 I will remove that for now, It would deserve it's own spec and it's an off-topic.